cmd/kubectl-directpv/init.go

```diff
@@ -0,0 +1,204 @@
+// This file is part of MinIO DirectPV
+// Copyright (c) 2021, 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/minio/directpv/pkg/admin"
+	"github.com/minio/directpv/pkg/consts"
+	"github.com/minio/directpv/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var errInitFailed = errors.New("init failed")
+
+var initCmd = &cobra.Command{
+	Use:           "init drives.yaml",
+	Short:         "Initialize the drives",
+	SilenceUsage:  true,
+	SilenceErrors: true,
+	Example: strings.ReplaceAll(
+		`# Initialize the drives
+$ kubectl {PLUGIN_NAME} init drives.yaml`,
+		`{PLUGIN_NAME}`,
+		consts.AppName,
+	),
+	Run: func(c *cobra.Command, args []string) {
+		if err := validateAdminServerConfigArgs(); err != nil {
+			utils.Eprintf(quietFlag, true, "%v\n", err)
+			os.Exit(-1)
+		}
+
+		switch len(args) {
+		case 1:
+		case 0:
+			utils.Eprintf(quietFlag, true, "Please provide the input file. Use `--help` for examples.")
+			os.Exit(-1)
+		default:
+			utils.Eprintf(quietFlag, true, "Too many input args. Use `--help` for examples.")
+			os.Exit(-1)
+		}
+
+		input := getInput(color.HiRedString("Initializing the drives will permanently erase existing data. Do you really want to continue (Yes|No)? "))
+		if input != "Yes" {
+			utils.Eprintf(quietFlag, false, "Aborting...\n")
+			os.Exit(1)
+		}
+
+		initMain(c.Context(), args[0])
+	},
+}
+
+func init() {
+	addAdminServerFlag(initCmd)
+}
+
+func constructInitDevicesRequest(config *InitConfigV1) admin.InitDevicesRequest {
```

Comment:
1. rename to `toInitDevicesRequest()`
2. Use non-versioned `InitConfig`  and respective structures everywhere.